examples/pytorch_lightning/swav_queue.py

```diff
@@ -0,0 +1,155 @@
+# Note: The model and training settings do not follow the reference settings
+# from the paper. The settings are chosen such that the example can easily be
+# run on a small dataset with a single GPU.
+
+import torch
+from torch import nn
+import torchvision
+import pytorch_lightning as pl
+
+from lightly.data import LightlyDataset
+from lightly.data import SwaVCollateFunction
+from lightly.loss import SwaVLoss
+from lightly.loss.memory_bank import MemoryBankModule
+from lightly.models.modules import SwaVProjectionHead
+from lightly.models.modules import SwaVPrototypes
+
+
+class SwaV(pl.LightningModule):
+    def __init__(
+        self,
+        num_ftrs,
+        out_dim,
+        n_prototypes,
+        n_queues,
+        queue_length=0,
+        start_queue_at_epoch=0,
+        n_steps_frozen_prototypes=0,
+    ):
+        super().__init__()
+        resnet = torchvision.models.resnet18()
+        self.backbone = nn.Sequential(*list(resnet.children())[:-1])
+        self.projection_head = SwaVProjectionHead(num_ftrs, num_ftrs, out_dim)
+        self.prototypes = SwaVPrototypes(
+            out_dim, n_prototypes, n_steps_frozen_prototypes
+        )
+        self.queues = None
+        if n_queues > 0:
+            self.queues = [MemoryBankModule(size=queue_length) for _ in range(n_queues)]
+            self.queues = nn.ModuleList(self.queues)
+            self.queue_length = queue_length
+            self.num_features_queued = 0
+            self.start_queue_at_epoch = start_queue_at_epoch
```

Comment:
I would ignore the case with no queues and just set it directly:

```suggestion
        self.start_queue_at_epoch = 15
        self.queues = nn.ModuleList([MemoryBankModule(size=3840) for _ in range(2)])
```